_posts/2024-12-04-enable-pluggable-storage-in-opensearch-vectordb.md

```diff
@@ -0,0 +1,255 @@
+
+![Overview of searchable snapshots](/assets/media/blog-images/2024-12-04-enable-pluggable-storage-in-opensearch-vectordb/searchable_snapshots_overview.png){:style="width: 100%; max-width: 600px;"}
+
+Follow these steps to configure searchable snapshots.
+
+### Prerequisites
+
+Configure your cluster for searchable snapshots. For detailed steps, see [Configuring a node to use searchable snapshots](https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/snapshots/searchable_snapshot/#configuring-a-node-to-use-searchable-snapshots).
+
+### Step 1: Create a local index
+
+Use the following request to create a local vector index:
+
+```json
+PUT /knn-index/
+{
+  "settings": {
+    "index": {
+      "knn": true
+    }
+  },
+  "mappings": {
+    "properties": {
+      "my_vector": {
+        "type": "knn_vector",
+        "dimension": 2
+      }
+    }
+  }
+}
+```
+
+### Step 2: Ingest data
+
+Ingest some data into the index:
+
+```json
+POST _bulk?refresh
+{ "index": { "_index": "knn-index", "_id": "1" } }
+{ "my_vector": [1.5, 2.5], "price": 12.2 }
+{ "index": { "_index": "knn-index", "_id": "2" } }
+{ "my_vector": [2.5, 3.5], "price": 7.1 }
+{ "index": { "_index": "knn-index", "_id": "3" } }
+{ "my_vector": [3.5, 4.5], "price": 12.9 }
+{ "index": { "_index": "knn-index", "_id": "4" } }
+{ "my_vector": [5.5, 6.5], "price": 1.2 }
+{ "index": { "_index": "knn-index", "_id": "5" } }
+{ "my_vector": [4.5, 5.5], "price": 3.7 }
+```
+
+### Step 3: Query the local index 
+
+Query the local index to ensure it's configured correctly:
+
+```json
+POST knn-index/_search
+{
+  "query": {
+    "knn": {
+      "my_vector": {
+        "vector": [2, 3],
+        "k": 2
+      }
+    }
+  }
+}
+```
+
+The response returns vectors closest to the query vector:
+
+```json
+{
+  "took": 16,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 2,
+      "relation": "eq"
+    },
+    "max_score": 0.6666667,
+    "hits": [
+      {
+        "_index": "knn-index",
+        "_id": "1",
+        "_score": 0.6666667,
+        "_source": {
+          "my_vector": [
+            1.5,
+            2.5
+          ],
+          "price": 12.2
+        }
+      },
+      {
+        "_index": "knn-index",
+        "_id": "2",
+        "_score": 0.6666667,
+        "_source": {
+          "my_vector": [
+            2.5,
+            3.5
+          ],
+          "price": 7.1
+        }
+      }
+    ]
+  }
+}
+```
+
+### Step 4: Take a snapshot
+
+Take a snapshot of the index. For detailed steps, see [Take and restore snapshots](https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore/). After taking the snapshot, delete the `knn-index` so it is no longer available locally.
```

Comment:
```suggestion
Take a snapshot of the index. For detailed steps, see [Take and restore snapshots](https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore/). After taking the snapshot, delete the `knn-index` so that it is no longer available locally.
```